src/metabase/util.cljc

```diff
@@ -917,3 +917,31 @@
               (map-all f (rest s1) (rest s2)))))))
   ([f c1 c2 & colls]
    (map-all* f (list* c1 c2 colls))))
+
+(defn row-diff
```

Comment:
This is very similar to `classify-changes`, another function in this namespace! This one is more flexible, but it would be tidier to have one function with optional args instead of two, or implement `classify-changes` in terms of `row-diff`. Alternatively, you could just use `classify-changes` and add a `get-id` argument.

At the very least, place these functions next to each other (and their tests too) so someone can clean them up later.